### PR TITLE
[270E][3.1.x] Update dependencies

### PR DESCRIPTION
**Pull request by jvega190**
Updates:
react: 16.8.6 -> 17.0.0
react-dom: 16.8.6 -> 17.0.0
react-scripts: 3.1.2 -> 4.0.3
typescript: 3.7.3 -> 4.1.2
babel-plugin-react-intl: 5.1.11 -> 5.1.16
node-sass: 4.13.1 -> 5.0.0

- Material UI updates (imports, props)
- Typescript errors/warning from new version 

https://github.com/craftersoftware/craftercms/issues/270

### CODE REVIEW

File: ui/app/src/components/BulkUpload.tsx [hunk 1]
```diff
@@ -562,12 +562,16 @@ const minimizedBarStyles = makeStyles((theme: Theme) =>
...
+  status?: {
+    status: string;
+    progress: number;
+  };
 
   onMaximized(): void;
 }
 
-function MinimizedBar(props: any) {
+// TODO: use type, status has a property status
```
Comment (rart) [inline ui/app/src/components/BulkUpload.tsx]: ?

File: ui/app/src/components/DependencySelection.tsx [hunk 2]
```diff
@@ -324,17 +324,8 @@ export function DependencySelectionDelete(props: DependencySelectionProps) {
...
-    items,
-    uris,
-    onItemClicked,
-    onSelectAllClicked,
-    displayItemTitle,
-    checked,
-    setChecked
-  } = props;
+  const { title, subtitle, items, uris, onItemClicked, onSelectAllClicked, displayItemTitle, checked, setChecked } =
+    props;
```
Comment (rart) [inline ui/app/src/components/DependencySelection.tsx:328]: Formatting doesn't look right.

File: static-assets/components/cstudio-admin/mods/admin-configurations.js [hunk 3]
```diff
@@ -175,12 +175,12 @@
           /**/ `<h2 class="hint--title">${formatMessage(adminConfigurationMessages.encryptMarked)}</h2>` +
           /**/ `<p>${formatMessage(adminConfigurationMessages.encryptHintPt1)}</p>` +
           /**/ `<p>` +
-          /**/ formatMessage(adminConfigurationMessages.encryptHintPt2, bold).join('') +
+          /**/ formatMessage(adminConfigurationMessages.encryptHintPt2, bold).props.children.join(' ') +
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/admin-configurations.js]: Think this is a dangerous. I've been holding updating react-intl for this reason. Let's discuss.

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-admin/mods/admin-configurations.js (hunk 3 above)
Comment (jvega190) [reply static-assets/components/cstudio-admin/mods/admin-configurations.js]: Reverted react-intl version

File: ui/app/src/components/DependencySelection.tsx (hunk 2 above)
Comment (jvega190) [reply ui/app/src/components/DependencySelection.tsx:328]: I ran prettier in the file and it says that is properly formatted

File: ui/app/package.json [hunk 4]
```diff
@@ -22,16 +22,15 @@
     "intl-messageformat": "^7.2.1",
     "js-cookie": "^2.2.1",
     "query-string": "^6.11.1",
-    "react": "^16.8.6",
+    "react": "17.0.0",
     "react-animations": "^1.0.0",
-    "react-dom": "^16.8.6",
+    "react-dom": "17.0.0",
```
Comment (rart) [inline ui/app/package.json]: Update to react@17.0.2

File: ui/app/package.json [hunk 5]
```diff
@@ -55,6 +54,38 @@
     "rules": {
       "prettier/prettier": [
         "error"
+      ],
+      "spaced-comment": [
+        "error",
+        "always",
+        {
+          "line": {
```
Comment (rart) [inline ui/app/package.json]: What were these added here for?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/package.json (hunk 5 above)
Comment (jvega190) [reply ui/app/package.json]: not needed, removing them

File: ui/app/package.json [hunk 6]
```diff
@@ -22,16 +22,15 @@
     "intl-messageformat": "^7.2.1",
     "js-cookie": "^2.2.1",
     "query-string": "^6.11.1",
-    "react": "^16.8.6",
+    "react": "^17.0.2",
     "react-animations": "^1.0.0",
-    "react-dom": "^16.8.6",
+    "react-dom": "17.0.0",
```
Comment (rart) [inline ui/app/package.json]: Upgrade to react 17.0.2

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**